pennylane/templates/subroutines/amplitude_amplification.py

```diff
@@ -0,0 +1,185 @@
+
+    Args:
+        U (Operator): Operator that generate the state :math:`|\Psi\rangle`.
+        O (Operator): The oracle that flips the sign of the state :math:`|\phi\rangle` and do nothing to the state :math:`|\phi^{\perp}\rangle`.
+        iters (int): the number of iterations of the amplitude amplification subroutine. Default is 1.
+        fixed_point (bool): whether to use the fixed-point amplitude amplification algorithm. Default is False.
+        work_wire (int): the auxiliary wire to use for the fixed-point amplitude amplification algorithm. Default is None.
+        reflection_wires (Wires): the wires to reflect on. Default is the wires of U.
+        p_min (int): the lower bound for the probability of success in fixed-point amplitude amplification. Default is 0.9
+
+    Raises:
+        ValueError: work_wire must be specified if fixed_point == True.
+        ValueError: work_wire must be different from the wires of U.
+
+    **Example**
+
+    Amplification of state :math:`|2\rangle` using Grover's algorithm with 3 qubits:
+
+    .. code-block::
+
+        @qml.prod
+        def generator(wires):
+          for wire in wires:
+            qml.Hadamard(wires = wire)
+
+        U = generator(wires = range(3))
+        O = qml.FlipSign(2, wires = range(3))
+
+        dev = qml.device("default.qubit")
+
+        @qml.qnode(dev)
+        def circuit():
+
+          generator(wires = range(3))
+          qml.AmplitudeAmplification(U, O, iters = 5, fixed_point=True, work_wire=3)
+
+          return qml.probs(wires = range(3))
+
+    .. code-block:: pycon
+
+        >>> print(np.round(circuit(),3))
+        [0.009 0.009 0.94  0.009 0.009 0.009 0.009 0.009]
+
+
+    """
+
+    def _flatten(self):
+        data = (self.hyperparameters["U"], self.hyperparameters["O"])
+        metadata = tuple(
+            value for key, value in self.hyperparameters.items() if key not in ["O", "U"]
+        )
+        return data, metadata
+
+    @classmethod
+    def _unflatten(cls, data, metadata):
+        U, O = (data[0], data[1])
+        return cls(
+            U,
+            O,
+            iters=metadata[0],
+            fixed_point=metadata[1],
+            work_wire=metadata[2],
+            p_min=metadata[3],
+            reflection_wires=metadata[4],
+        )
+
+    def __init__(
+        self, U, O, iters=1, fixed_point=False, work_wire=None, p_min=0.9, reflection_wires=None
+    ):
+        self._name = "AmplitudeAmplification"
+        if reflection_wires is None:
+            reflection_wires = U.wires
+
+        if fixed_point and work_wire is None:
+            raise qml.wires.WireError("work_wire must be specified if fixed_point == True.")
+
+        if fixed_point and len(U.wires + qml.wires.Wires(work_wire)) == len(U.wires):
+            raise ValueError("work_wire must be different from the wires of U.")
```

Comment:
Yes, what I am doing is adding work_wire to the wires of U. The wires are treated as sets so if the set size does not increase it is because it is a subset. I tried first:

`qml.wires.Wires(work_wire) in U.wires`

or just

`work_wire in U.wires`

but it gave problems in certain examples